Extending/Section-Trees/Searchable-Trees/index-v10.md

```diff
@@ -0,0 +1,167 @@
+---
+versionFrom: 10.0.0
+---
+
+# Searchable Trees (ISearchableTree)
+
+When you type a search term into the Umbraco backoffice search field, you'll see search results from all the Section Trees that your user account has permission to access:
+
+![Content Section Dashboards](images/backoffice-search-v8.png)
+
+The results are grouped by 'Section Tree' e.g. Content, Media, Document Types: each 'Tree' has its own associated search mechanism that receives the search term and looks for matches in the tree that is responsible for searching.
+
+You can create your own search mechanisms for your own custom sections or replace the default search implementation for a particular section tree.
+
+## Custom Tree Search
+
+To create a search for your own custom tree you need to create a C# class that implements the interface `Umbraco.Cms.Core.Trees.ISearchableTree`.
+
+### ISearchableTree
+
+```csharp
+using System.Collections.Generic;
+using Umbraco.Cms.Core.Composing;
+using Umbraco.Cms.Core.Models.ContentEditing;
+
+namespace My.Website
+{
+    public interface ISearchableTree : IDiscoverable
+    {
+        /// <summary>
+        /// The alias of the tree that the <see cref="ISearchableTree"/> belongs to
+        /// </summary>
+        string TreeAlias { get; }
+
+        /// <summary>
+        /// Searches for results based on the entity type
+        /// </summary>
+        /// <param name="query">The search term used for finding matching results.</param>
+        /// <param name="pageSize">The number of records to return for a page of results.</param>
+        /// <param name="pageIndex">The 0-based index for retrieving a page of search results.</param>
+        /// <param name="totalFound">Populated with the total number of results matching the provided search term.</param>
+        /// <param name="searchFrom">
+        ///     The starting point for the search, generally a node ID, but for members this is a member type alias.
+        /// </param>
+        /// <returns></returns>
+        IEnumerable<SearchResultEntity> Search(string query, int pageSize, long pageIndex, out long totalFound, string searchFrom = null);
+    }
+}
+```
+
+Your implementation needs to return an IEnumerable of `SearchResultEntity` items:
+
+```csharp
```

Comment:
```
using Umbraco.Cms.Core.Models.ContentEditing;
using System.Runtime.Serialization;

namespace My.Website
{
    public class SearchResultEntity : EntityBasic
    {
        public SearchResultEntity() { }
        /// <summary>
        /// The score of the search result
        /// </summary>
        [DataMember(Name = "score")]
        public float Score { get; set; }
    }
}
```

SearchResultEntity() throws an error "Must declare a body.."

Reply:
I am not entirely sure I understand what you mean here? 🤔

Reply:
Ah, OK.

I'll run it by the community team to double-check it 💪